lib/letter.rb

```diff
@@ -164,4 +194,13 @@ def strip_html(html)
 
     return html
   end
+
+  def xpath_parent xpath
+    return @xml.at_xpath(xpath.partition(/\/\/tei\:\w+\[\d+\]$/)[0], "tei" => $tei_ns)
```

Comment:
I'm really sad that Nokogiri doesn't just have an xPath class which we could be using with methods like `.siblings` and `.parent` :(